core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -190,8 +193,35 @@ public RelFieldCollation toFieldCollation(Map<String, Object> map) {
     return new RelFieldCollation(field, direction, nullDirection);
   }
 
-  public RelDistribution toDistribution(Object o) {
-    return RelDistributions.ANY; // TODO:
+  public RelDistribution toDistribution(Map<String, Object> map) {
+    final RelDistribution.Type type =
```

Comment:
Is it a break change?

Reply:
In Apache Hive there is a command `EXPLAIN FORMATTED <query>`: it prints out the json representation of the plan generated by Calcite.
I am working on an implementation of enabling Calcite planner in Hive when queries has `SORT BY` clause. `SORT BY` is translated to `SortExchange` which has `RelDistribution` and the command I mentioned above failed with serialization exception.
This change is part of a bigger one: https://issues.apache.org/jira/browse/HIVE-22785

Reply:
@chunweilei I don't think this is a breaking change - this `RelJson` class is only used from `RelJsonReader` - and it doesn't seem valueable outside of that (note: I think `RelJson` could be viewed as an implementation detail and as such it could be made package private)

Reply:
Every public method is a public API. But I don't have a strong objection about it since it is not a frequently used api.